### PR TITLE
Create a build for the Controller

### PR DESCRIPTION
**Pull request by mitchellhamilton**
This fixes a problem that @ticidesign found when creating custom field types outside the monorepo.

What happened is that the Controller file which field type Controllers extend isn't compiled since the webpack config(intentionally) won't compile anything in node_modules so this adds another preconstruct entrypoint to the fields package for the Controller file.

Note that this has a major changeset to the fields package just in case people imported `@keystone-alpha/fields/Controller.js`(explicitly with the `.js`).

### CODE REVIEW

**Comment by mitchellhamilton** [comment]
> Why packages/fields/Controller-src.js and not packages/fields/Controller/index.js?

That would work but I think it would be slightly confusing because people would have to know the specifics of the order which things happen in the node module resolution algorithm and that for the fields package, people would have to import `../some number of ../Controller/index` and `../some number of ../Controller` wouldn't work and for outside the fields package, the opposite would be true. Whereas having `src` in there something, it's more clear what should be imported where.

Do you think packages/fields/Controller/src/index.js would be better?

**Comment by gautamsi** [comment]
is this because of `Flow`? Does Flow support different file extension? like `.flow` or `.flow.js` so that we do not have to have `controller-src.js` and compiled `controller.js`.

I did bring `TypeScript` in past. I know that is not favorable here.   TypeScript has its extension `.ts`. 
`TypeScript` also works like Flow too with purely based on jsdoc comments in js files without the need of compiling them again (no typescript specific stuffs in js file). It does work with `babel 7` so no typescript step needed outside of existing compile step. 

There are several issues related to file extension in flow repo.  one of them is https://github.com/facebook/flow/issues/4233

**Comment by jesstelford** [comment]
> _ and for outside the fields package, the opposite would be true. Do you think packages/fields/Controller/src/index.js would be better?_

It's already different between the internal and external imports. I'd be happier with `packages/fields/Controller/src/index.js` - it's more explicit, and I've seen that general pattern a lot in the node ecosystem, so is easy to understand.

File: packages/fields/Controller/package.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "main": "dist/fields.cjs.js",
+  "module": "dist/fields.esm.js",
+  "preconstruct": {
+    "source": "../Controller-src.js"
```
Comment (jesstelford) [inline packages/fields/Controller/package.json]:
```suggestion
    "source": "../Controller/src/index.js"
```

File: packages/fields/types/CalendarDay/views/Controller/Controller.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/CalendarDay/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Checkbox/views/Controller/Controller.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Checkbox/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/DateTime/views/Controller/Controller.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/DateTime/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Decimal/views/Controller/Controller.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Decimal/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/File/views/Controller/Controller.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable import/no-extraneous-dependencies */
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/File/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Float/views/Controller/Controller.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Float/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Integer/views/Controller/Controller.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Integer/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Password/views/Controller/Controller.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable import/no-extraneous-dependencies */
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Password/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Relationship/views/Controller/Controller.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Relationship/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Select/views/Controller/Controller.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Select/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

File: packages/fields/types/Text/views/Controller/Controller.js [hunk 12]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import FieldController from '../../../../Controller';
+import FieldController from '../../../../Controller-src';
```
Comment (jesstelford) [inline packages/fields/types/Text/views/Controller/Controller.js]:
```suggestion
import FieldController from '../../../../Controller/src/index';
```

**Comment by jesstelford** [comment]
@mitchellhamilton I've gone ahead and made those changes on this branch 👍

**Pull request merged**